src/Bravo3/Orm/Mappers/Yaml/YamlMapWriter.php

```diff
@@ -0,0 +1,332 @@
+     * @return array
+     */
+    private function compileStdIndices(Entity $md)
+    {
+        $out     = [];
+        $indices = $md->getIndices();
+
+        foreach ($indices as $index) {
+            $data = [];
+
+            if ($index->getColumns()) {
+                $data[Schema::INDEX_COLUMNS] = $index->getColumns();
+            }
+
+            if ($index->getMethods()) {
+                $data[Schema::INDEX_METHODS] = $index->getMethods();
+            }
+
+            $out[$index->getName()] = $data;
+        }
+
+        return $out;
+    }
+
+    /**
+     * Compile all sortable table indices
+     *
+     * @param Entity $md
+     * @return array
+     */
+    private function compileSortIndices(Entity $md)
+    {
+        return $this->compileSortables($md->getSortables());
+    }
+
+    /**
+     * Compile sortables into an array
+     *
+     * @param Sortable[] $sortables
+     * @return array
+     */
+    private function compileSortables(array $sortables)
+    {
+        $out = [];
+
+        foreach ($sortables as $index) {
+            $data = [];
+
+            if ($index->getColumn()) {
+                $data[Schema::INDEX_COLUMN] = $index->getColumn();
+            }
+
+            $conditions = $this->compileConditions($index->getConditions());
+            if ($conditions) {
+                $data[Schema::INDEX_CONDITIONS] = $conditions;
+            }
+
+            $out[$index->getName()] = $data;
+        }
+
+        return $out;
+    }
+
+    /**
+     * Compile conditions into an array
+     *
+     * @param \Bravo3\Orm\Mappers\Metadata\Condition[] $conditions
+     * @return array
+     */
+    private function compileConditions(array $conditions)
+    {
+        $out = [];
+
+        foreach ($conditions as $condition) {
+            $data = [Schema::CONDITION_VALUE => $condition->getValue()];
+
+            if ($condition->getColumn()) {
+                $data[Schema::CONDITION_COLUMN] = $condition->getColumn();
+            }
+
+            if ($condition->getMethod()) {
+                $data[Schema::CONDITION_METHOD] = $condition->getMethod();
+            }
+
+            if ($condition->getComparison() && ($condition->getComparison() != '=')) {
+                $data[Schema::CONDITION_COMPARISON] = $condition->getComparison();
+            }
+
+            $out[] = $data;
+        }
+
+        return $out;
+    }
+
+    /**
+     * Compiles all columns for an entity
+     *
+     * @param Entity $md
+     * @return array
+     */
+    private function compileColumns(Entity $md)
+    {
+        $out     = [];
+        $columns = $md->getColumns();
+
+        foreach ($columns as $column) {
+            $data = [
+                Schema::COLUMN_TYPE => $column->getType()->value(),
+            ];
+
+            if ($column->isId()) {
+                $data[Schema::COLUMN_ID] = true;
+            }
+
+            $default_getter = 'get'.Inflector::classify($column->getProperty());
```

Comment:
Yeah Tony was telling me about that at lunch - I think changing the inflector might count as another PR, though.